vaadin-notification-flow-parent/vaadin-notification-flow/src/main/java/com/vaadin/flow/component/notification/Notification.java

```diff
@@ -524,6 +524,15 @@ public Registration addDetachListener(
         return super.addDetachListener(listener);
     }
 
+    @Override
+    protected void onDetach(DetachEvent detachEvent) {
+        super.onDetach(detachEvent);
+
+        // If the notificaiton gets detached, it needs to be marked as closed so
+        // it won't auto-open when reattached.
+        setOpened(false);
```

Comment:
This will trigger the `openedChanged` listener above, right?

Reply:
That was the intention but it seems there was some timing issue, so I needed to change the logic. Let's see if the newest revision passes.